File: int/bin.cpp
#include <string>
#include <iostream>

using namespace std;

string bin(int n) {
  string ans = "";
  
  while (n) {
    ans = ((n % 2 == 1) ? '1' : '0') + ans;
    n >>= 1;
    n &= ~(1 << 31);
  }
  return ans;
}

int main() {
  for (int i = -10; i < 50; i++) {
    cout << bin(i) << endl;
  }
  return 0;
}

File: int/bin.py
import math


def binn(n):
    ans = []

    while n:
        print n, bin(n)
        ans.append('1' if n % 2 == 1 else '0')
        n = (n >> 1) ^ (1 << 32)
    return ''.join(reversed(ans))


for i in xrange(-10, 50):
    print binn(i)
